### PR TITLE
[Y360CMS-50] GroupExp Pro API spec

### PR DESCRIPTION
**Pull request by hamrant**

### CODE REVIEW

File: vc.adoc [hunk 1]
```diff
@@ -1076,6 +1087,152 @@ Content-Type: application/json;charset=UTF-8
...
+
+[width="100%",cols="25%,25%,25%,25%",options="header",]
+|===
+| Path | Type | Optional | Description
+| items | Map | false | Array of Key-Value to represent assets.
+| items[].groupexp_id | String | false | GroupExp object id.
+| items[].images | Array | false | Array of image objects.
+| items[].images[].format | String | false | Image style of the image file.
+| items[].images[].sourceUrl | String | false | Image object url of the image file.
+| items[]video |Object |false |Video brief info.
```
Comment (AndreyMaximov) [inline vc.adoc]: dot is missed

File: vc.adoc [hunk 2]
```diff
@@ -1076,6 +1087,152 @@ Content-Type: application/json;charset=UTF-8
...
+| items[]video.name |String |false |Name of the Video (e.g. 'RPM #79 Express', 'BODYPUMP #100').
```
Comment (AndreyMaximov) [inline vc.adoc]: dot is missed

File: vc.adoc (hunk 1 above)
Comment (AndreyMaximov) [reply vc.adoc]: Video is optional

File: vc.adoc [hunk 3]
```diff
@@ -1076,6 +1087,152 @@ Content-Type: application/json;charset=UTF-8
...
+| items[].id |String |false |Unique ID of the Video.
+| items[].duration |Integer |true |Video duration in seconds.
```
Comment (AndreyMaximov) [inline vc.adoc]: items[].duration -> items[].video.duration
Duration is mandatory

File: vc.adoc [hunk 4]
```diff
@@ -1076,6 +1087,152 @@ Content-Type: application/json;charset=UTF-8
...
+|===
+| Path | Type | Optional | Description
+| items | Map | false | Array of Key-Value to represent assets.
+| items[].groupexp_id | String | false | GroupExp object id.
+| items[].images | Array | false | Array of image objects.
+| items[].images[].format | String | false | Image style of the image file.
+| items[].images[].sourceUrl | String | false | Image object url of the image file.
+| items[]video |Object |false |Video brief info.
+| items[]video.name |String |false |Name of the Video (e.g. 'RPM #79 Express', 'BODYPUMP #100').
+| items[].id |String |false |Unique ID of the Video.
```
Comment (AndreyMaximov) [inline vc.adoc]: items[].id -> items[].video.id

File: vc.adoc [hunk 5]
```diff
@@ -1076,6 +1087,152 @@ Content-Type: application/json;charset=UTF-8
...
+| items[].duration |Integer |true |Video duration in seconds.
+| items[].episode |Object |true |Video Episode.
+| items[].episode.number |Integer |true |Number of episode in season.
+| items[].episode.season |Integer |true |Number of season.
+| items[].thumbnail |String |false |Thumbnail of the Video.
+| items[].customInfo |Map |false |Array of Key-Value to represent additional information of an entity. Partners could contain diverse information of same entities. This type designed to store that information.
```
Comment (AndreyMaximov) [inline vc.adoc]: `.video.` is missed in the lines above

File: vc.adoc [hunk 6]
```diff
@@ -1076,6 +1087,152 @@ Content-Type: application/json;charset=UTF-8
...
+            {
+               "format":"medium",
+               "sourceUrl":"http://master.y360cms.demo.fivejars.com/sites/default/files/styles/medium/public/downtown_exterior_0.jpg?itok=yaosJs2t"
+            },
+            {
+               "format":"small",
+               "sourceUrl":"http://master.y360cms.demo.fivejars.com/sites/default/files/styles/small/public/downtown_exterior_0.jpg?itok=yaosJs2t"
+            }
+         ],
+         "video":{
```
Comment (AndreyMaximov) [inline vc.adoc]: Let's incorporate a response example, where optional attributes are not presented

File: vc.adoc [hunk 7]
```diff
@@ -1076,6 +1087,152 @@ Content-Type: application/json;charset=UTF-8
...
+            "id":"7",
+            "name":"FREESTYLE SWIMMING - 5 MOST COMMON MISTAKES (2019)",
+            "thumbnail":"http://embed.wistia.com/deliveries/1a4efc93b070c0af918129bd86157661d70754dd.bin",
+            "duration":835,
+            "customInfo":[]
+         }
+      },
+      {
+         "groupexp_id":"9204039012021",
```
Comment (AndreyMaximov) [inline vc.adoc]: Let's have the id as a key, so that the response array is indexed and it's easier to get the needed object.

File: vc.adoc (hunk 6 above)
Comment (hamrant) [reply vc.adoc]: fixed

File: vc.adoc (hunk 7 above)
Comment (hamrant) [reply vc.adoc]: fixed

File: vc.adoc [hunk 8]
```diff
@@ -1076,6 +1087,145 @@ Content-Type: application/json;charset=UTF-8
...
+[[_example_response_group_exp]]
+==== link:#_example_response_group_exp[Example response]
+
+[source,highlightjs,highlight,nowrap]
+----
+HTTP/1.1 200 OK
+Content-Length: 417
+Content-Type: application/json;charset=UTF-8
+
+{
```
Comment (m-titov) [inline vc.adoc]: Could you please change to:

```
{
  "items": [
    {
      "groupexp_id": "9204039012020",
      "images": [
        {
          "format": "large",
          "sourceUrl": "http://master.y360cms.demo.fivejars.com/sites/default/files/styles/large/public/downtown_exterior_0.jpg?itok=yaosJs2t"
        },
        {
          "format": "medium",
          "sourceUrl": "http://master.y360cms.demo.fivejars.com/sites/default/files/styles/medium/public/downtown_exterior_0.jpg?itok=yaosJs2t"
        },
        {
          "format": "small",
          "sourceUrl": "http://master.y360cms.demo.fivejars.com/sites/default/files/styles/small/public/downtown_exterior_0.jpg?itok=yaosJs2t"
        }
      ],
      "video": [
        {
          "id": "7",
          "name": "FREESTYLE SWIMMING - 5 MOST COMMON MISTAKES (2019)",
          "thumbnail": "http://embed.wistia.com/deliveries/1a4efc93b070c0af918129bd86157661d70754dd.bin",
          "duration": 835,
          "customInfo": []
        }
      ]
    },
    {
      "groupexp_id": "9204039012021",
      "images": [
        {
          "format": "large",
          "sourceUrl": "http://master.y360cms.demo.fivejars.com/sites/default/files/styles/large/public/downtown_exterior_0.jpg?itok=yaosJs2t"
        },
        {
          "format": "medium",
          "sourceUrl": "http://master.y360cms.demo.fivejars.com/sites/default/files/styles/medium/public/downtown_exterior_0.jpg?itok=yaosJs2t"
        },
        {
          "format": "small",
          "sourceUrl": "http://master.y360cms.demo.fivejars.com/sites/default/files/styles/small/public/downtown_exterior_0.jpg?itok=yaosJs2t"
        }
      ]
    }
  ]
}
```?

File: vc.adoc [hunk 9]
```diff
@@ -1076,6 +1087,145 @@ Content-Type: application/json;charset=UTF-8
...
...e":"Step Express 2", "type":null, "startDateTime":1579532400000, "endDateTime":1579534200000, "maxCapacity":null, "totalBooked":null, "waitlistCapacity":null, "waitlistBooked":null, "availableSpots":null, "instructor":{ "id":"Heidi R", "fullName":"Heidi R", "customInfo":null }, "activity":{ "id":"Step Express", "description":"Step Express" }, "free":true, "childCare":false, "reservable":false, "availableOptions":null, "cancelled":false, "booked":false, "waitlisted":null, "customInfo":null, "clubUuid":"764e2227-62d1-4a02-91a5-21bc52cc5503" }, "details":{ "id":null, "description":"Everything you love about step in a shorter time frame!", "webCapacity":null, "webBooked":null, "room":{ "id":"C", "description":"C", "roomPhotoUrl":null, "customInfo":null }, "level":null, "pricing":null, "legalNotes":null, "additionalInfo":null, "customInfo":null, "cancellationWindowEnd":null, "bookingWindowStart":null, "bookingWindowEnd":null, "imageUrl":null, "videoUrl":null }, "attendeeDetails":{ "exerciserUuid":"d7b35f68-fa63-4db3-8764-5465be8ae11b", "booked":false, "spotBooked":null, "waitlistBooked":false, "waitlistPosition":null, "productAvailability":"NOT_APPLICABLE", "availableActions":[ ] } } }'
```
Comment (m-titov) [inline vc.adoc]: Could we send only  class id, class name and class description in request ?

File: vc.adoc (hunk 8 above)
Comment (hamrant) [reply vc.adoc]: fixed

File: vc.adoc [hunk 10]
```diff
@@ -1076,6 +1087,163 @@ Content-Type: application/json;charset=UTF-8
...
+
+[width="100%",cols="25%,25%,25%,25%",options="header",]
+|===
+|Parameter |Type |Optional |Description
+|body |String |false |GroupExp objects in JSON format.
+|===
+Example:
+----
+{
+   "provider":"GroupExp v1",
```
Comment (m-titov) [inline vc.adoc]: I would like to rename it somehow. For example, `source`

File: vc.adoc [hunk 11]
```diff
@@ -1076,6 +1087,163 @@ Content-Type: application/json;charset=UTF-8
...
+| items[].video[].thumbnail |String |false |Thumbnail of the Video.
+| items[].video[].customInfo |Map |false |Array of Key-Value to represent additional information of an entity. Partners could contain diverse information of same entities. This type designed to store that information.
+|===
+
+[[_example_request_group_exp]]
+==== link:#_example_request_group_exp[Example request]
+
+[source,highlightjs,highlight]
+----
+$ curl 'http://localhost:8080/api/groupexp/v1.0/content-providers/wichita/get-category-asset?exerciserUuid=8965a460-a79e-4bf7-b66c-7e34d8c34760' -i -X POST \
```
Comment (m-titov) [inline vc.adoc]: Please replace `groupexp`  with `groupxclasses` or remove.

Comment (m-titov) [reply vc.adoc]: And I would like to replace it with `GET /api/groupxclasses/v1.0/content-providers/{provider}/get-category-asset?source={source}&id={id}&name={name}&activity={activity}`

File: vc.adoc (hunk 10 above)
Comment (hamrant) [reply vc.adoc]: fixed

File: vc.adoc (hunk 11 above)
Comment (hamrant) [reply vc.adoc]: fixed

**Pull request merged**